pkg/logging/http.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package logging
+
+import (
+	"fmt"
+
+	"net/http"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	package_http "github.com/thanos-io/thanos/pkg/server/http"
+)
+
+type ServerMiddleware struct {
+	opts   *options
+	logger log.Logger
+}
+
+func (c *ServerMiddleware) PostCall(name string, start time.Time, wrapped *package_http.ResponseWriter, r *http.Request) {
+	logger := log.With(c.logger, "http.request_logging_server", name)
+
+	reqID := r.Header.Get("X-Request-ID")
+	logger = log.With(logger, "http.request_id", reqID)
+	status := wrapped.Status()
+
+	if status >= 200 && status < 500 {
+		logger = log.With(logger, "http.status_code", fmt.Sprintf("%d", status))
+	}
+	// server error, log them.
+	if status >= 500 && status < 600 {
+		logger = log.With(logger, "http.server_error", fmt.Sprintf("http.status_code: %v", status))
+	}
+
+	logger = log.With(logger, "http.time_ms", fmt.Sprintf("%v", durationToMilliseconds(time.Since(start))))
+	logger = c.opts.levelFunc(logger, status)
+	logger.Log("finished_http_call", name)
```

Comment:
It would be amazing to have consistent error messages for this. This suddenly uses `_` ? Can we have consistency between this and grpc one?

Reply:
In fact the best would be to  leverage structure logging .

We could have `protocol=http` and then just `finished call` and gRPC would have `protcol=gRPC` so we can filter and group things on log external system (: WDYT?

Reply:
Sure, infact I used `_` because there was no space between the words

> We could have protocol=http and then just finished call and gRPC would have protcol=gRPC so we can filter and group things on log external system (: WDYT?

Yup, we can add a `protocol` field, makes it more elegant.

Reply:
Added, can you take a look?

Reply:
Updated the convention in the grpc logging also - https://github.com/grpc-ecosystem/go-grpc-middleware/pull/311